daemon/CryptoImpl/crypto.cpp

```diff
@@ -323,6 +323,36 @@ void Daemon::ApiImpl::CryptoDBusObject::decrypt(
                                   result);
 }
 
+void Daemon::ApiImpl::CryptoDBusObject::authenticatedDecrypt(
+        const QByteArray &data,
+        const QByteArray &iv,
+        const Key &key,
+        CryptoManager::BlockMode blockMode,
+        CryptoManager::EncryptionPadding padding,
+        const QByteArray &authenticationData,
+        const QByteArray &tag,
+        const QString &cryptosystemProviderName,
+        const QDBusMessage &message,
+        Result &result,
+        QByteArray &decrypted)
+{
+    Q_UNUSED(decrypted);  // outparam, set in handlePendingRequest / handleFinishedRequest
+    QList<QVariant> inParams;
+    inParams << QVariant::fromValue<QByteArray>(data);
```

Comment:
Not a problem, just a question: is the template argument necessary here? Can't the compiler figure it out on its own without explicity specifying it?

Reply:
It's not necessary for Qt types like QByteArray and QString; it used to be required for custom registered types, but I'm not sure of the requirements for current Qt/compiler configurations.